crates/xtask/src/download_regression_tests.rs

```diff
@@ -0,0 +1,137 @@
+use std::path::PathBuf;
+use std::process::Command;
+
+pub(crate) fn download_regression_tests() -> Result<()> {
+    let target_dir = PathBuf::from("crates/squawk_parser/tests/data/regression_suite");
+
+    if target_dir.exists() {
+        println!("Cleaning target directory: {}", target_dir.display());
+        remove_dir_all(&target_dir)?;
+    }
+
+    create_dir_all(&target_dir)?;
+
+    let urls = fetch_download_urls()?;
+    let total_files = urls.len();
+
+    for (index, url) in urls.iter().enumerate() {
+        let filename = url.split('/').last().unwrap();
+        let filepath = target_dir.join(filename);
+
+        println!(
+            "[{}/{}] Downloading {}... ",
+            index + 1,
+            total_files,
+            filename
+        );
+
+        let output = Command::new("curl").args(["-s", url]).output()?;
+
+        if !output.status.success() {
+            let error_msg = String::from_utf8_lossy(&output.stderr);
+            return Err(anyhow::anyhow!(
+                "Failed to download '{}': {}",
+                url,
+                error_msg
+            ));
+        }
+
+        let mut processed_content = Vec::new();
+
+        let cursor = Cursor::new(&output.stdout);
+
+        if let Err(e) = preprocess_sql(cursor, &mut processed_content) {
+            eprintln!("Error: Failed to process file: {}", e);
+            continue;
+        }
+
+        let mut dest = File::create(&filepath)?;
+        dest.write_all(&processed_content)?
+    }
+
+    Ok(())
+}
+
+fn fetch_download_urls() -> Result<Vec<String>> {
+    // Fetch list of SQL file URLs
+    println!("Fetching SQL file URLs...");
+    let output = Command::new("gh")
+        .args([
+            "api",
+            "-H",
+            "Accept: application/vnd.github+json",
+            "/repos/postgres/postgres/contents/src/test/regress/sql",
+        ])
+        .output()?;
+
+    if !output.status.success() {
+        return Err(anyhow::anyhow!(
+            "Failed to fetch SQL files: {}",
+            String::from_utf8_lossy(&output.stderr)
+        ));
+    }
+
+    let json_str = String::from_utf8(output.stdout)?;
+    let files: Vec<serde_json::Value> = serde_json::from_str(&json_str)?;
+
+    // Extract download URLs for SQL files
+    let urls: Vec<String> = files
+        .into_iter()
+        .filter(|file| {
+            file["name"]
+                .as_str()
+                .map(|name| name.ends_with(".sql"))
+                .unwrap_or(false)
+        })
+        .filter_map(|file| file["download_url"].as_str().map(String::from))
+        .collect();
+
+    if urls.is_empty() {
+        return Err(anyhow::anyhow!("No SQL files found"));
+    }
+
+    Ok(urls)
+}
+
+fn preprocess_sql<R: BufRead, W: Write>(source: R, mut dest: W) -> Result<()> {
+    let mut skipping_copy_block = false;
+
+    for line in source.lines() {
+        let mut line = line?;
+
+        // Detect the start of the COPY block
```

Comment:
what does a copy block look like? curious what it has that we don't support parsing

Reply:
the regression test is executed via `psql`, and they use `COPY` to load data into test tables `FROM STDIN`

```sql
COPY bitwise_test FROM STDIN NULL 'null';
1	1	1	1	1	B0101
3	3	3	null	2	B0100
7	7	7	3	4	B1100
\.
```